src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -20,6 +20,32 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Private
 
+;; TODO should these IllegalArgumentExceptions be schema errors?
+(defn ^:private reject! [& strings]
+  (throw (IllegalArgumentException. (apply str strings))))
+
+(schema/defn validate-ace-config-map!
+  [config-value :- (schema/pred map? "map?")]
+  (when (:attrs config-value)
+    (schema/validate acl/CSRAttributes (:attrs config-value)))
+
+  (when (:certname config-value)
+    (schema/validate String config-value))
+
+  (when (or (not (or (:attrs config-value) (:certname config-value)))
```

Comment:
Might be better to do this with a schema predicate? Then our error handling would be consistent across this function